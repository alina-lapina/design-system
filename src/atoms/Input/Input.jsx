import React, { useState } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';
import { AlertTriangle, Search }Â from 'react-feather';

const Input = ({
	disabled, error, errorMessage, handleChange, label, negative, searchField, submitCallback, type, value,
}) => {
	const [inputValue, setValue] = useState(value);
	const id = uuid();
	const handleInputChange = e => {
		setValue(e.target.value);
		handleChange(e.target.value);
	};

	return (
		<div className={`input-field-wrapper ${negative ? 'negative' : 'standard'}${error ? ' with-error' : ''}`}>
			<label htmlFor={id}>{label}</label>
			<div className="d-flex">
				<input
					className={searchField ? 'has-icon' : ''}
					id={id}
					disabled={disabled}
					type={type}
					value={inputValue}
					onChange={e => handleInputChange(e)}
				/>
				{searchField && (
					<div className="input-icon" onClick={() => submitCallback(inputValue)}>
						<Search className="search-icon" size={18} />
					</div>
				)}
				{error && (
					<div className="input-icon">
						<AlertTriangle className="alert-icon" size={18} />
					</div>
				)}
			</div>
			{error && (errorMessage && (
				<span className="error-message roboto">{ errorMessage }</span>
			))}
		</div>
	);
};

Input.defaultProps = {
	disabled: false,
	error: false,
	handleChange: () => {},
	label: 'Input field',
	negative: false,
	searchField: false,
	submitCallback: () => {},
	type: 'text',
};

Input.propTypes = {
	disabled: PropTypes.bool,
	error: PropTypes.bool,
	errorMessage: PropTypes.string,
	handleChange: PropTypes.func,
	label: PropTypes.string,
	negative: PropTypes.bool,
	searchField: PropTypes.bool,
	submitCallback: PropTypes.func,
	type: PropTypes.string,
	value: PropTypes.string,
};

export default Input;
