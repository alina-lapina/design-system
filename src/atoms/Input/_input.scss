@mixin activate-label {
  font-size: 14px;
  padding: 0 2px;
  top: -8px;
}

.input-field-wrapper {
  align-items: center;
  cursor: text;
  display: flex;
  margin: $global-padding-size;
  position: relative;

  &:hover {
    input {
      border: 1px solid $ssb-green-4;
      //outline: $ssb-green-4 auto 5px;
    }
  }

  input {
    border: 1px solid $ssb-dark-5;
    font-size: 16px;
    height: $input-field-height;
    padding: 4px $global-padding-size;
    text-overflow: ellipsis;
    width: 100%;

    &:disabled {
      border: 1px solid $ssb-dark-5;
      cursor: not-allowed;
    }

    &:disabled + label {
      color: rgba($label-color, .5);
      cursor: not-allowed;
      user-select: none;
    }

    &:focus {
      //border: 2px solid $ssb-green-4;
      outline: $ssb-green-4 auto 5px;
    }

    &:focus + label {
      @include activate-label;
      color: $ssb-green-4;
    }

    &.has-icon {
      padding-right: $input-field-height;
    }
  }

  label {
    background: $ssb-white;
    color: $label-color;
    cursor: text;
    left: $global-padding-size;
    position: absolute;
    top: 8px;
    transition: all .2s;
    user-select: none;

    &.has-value {
      @include activate-label;
    }
  }

  .clickable-button {
    align-items: center;
    cursor: pointer;
    display: flex;
    height: $input-field-height;
    margin-left: -$input-field-height;
    width: $input-field-height;

    .search-icon {
      color: $ssb-green-3;
    }
  }

}