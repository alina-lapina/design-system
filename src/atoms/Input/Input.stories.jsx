import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Input from './Input';

let someValue = '';
const handleChange = e => {
	someValue = e;
};

storiesOf('Atoms|Input', module).addDecorator(centered)
	.add('Default', () => (
		<Input value={someValue} handleChange={handleChange} />
	))
	.add('With value', () => (
		<Input value="Already filled" />
	))
	.add('Disabled', () => (
		<Input disabled />
	));
