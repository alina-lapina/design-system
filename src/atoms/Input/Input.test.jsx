import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Input />);
		expect(wrapper).toMatchSnapshot();
	});
});
