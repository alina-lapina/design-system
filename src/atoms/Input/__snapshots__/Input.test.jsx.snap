// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Matches the snapshot 1`] = `
<div
  className="input-field-wrapper"
>
  <input
    disabled={false}
    id="7df9fba2-b72d-472b-9d7b-071e92ee5648"
    onChange={[Function]}
    type="text"
  />
  <label
    className="undefined"
    htmlFor="7df9fba2-b72d-472b-9d7b-071e92ee5648"
  >
    Input field
  </label>
</div>
`;
